### DISCUSSION TITLE
Does this project have some standard python notebooks for embeddings, vector DB, etc?

### DISCUSSION THREAD

**Discussion by MaratZakirov** [ggml-org/llama.cpp#2386] [Q&A]
Recently I found some examples for embeddings. For example I found that `./embd-input/embd_input.py` should give me some vectors for query I guess? (I still trying to figure that out). It would be very helpful if there be some standard python notebooks for:
1. For embeddings
2. Making some sort of vector DB
3. Search engines based on embeddings
4. Some word2vec stuff based on embeddings
5. Some template examples of how context should be loaded to LLM in order to obtain quality outputs.

Or maybe I am missing something and all that examples are already exists in project?

**Comment by Technetium1** [comment]
What you may be missing is that there is already web chat available. Notebook usage is not something this project has historically encouraged. Notebooks are _not_ standard Python, even though they are popular right now. They are a part of [Project Jupyter](https://jupyter.org/). If you would like to use Python bindings, there is https://github.com/abetlen/llama-cpp-python, and it's available in langchain as well: https://python.langchain.com/docs/integrations/llms/llamacpp
